op-batcher/metrics/metrics.go

```diff
@@ -349,6 +353,13 @@ func (m *Metrics) RecordChannelQueueLength(len int) {
 	m.channelQueueLength.Set(float64(len))
 }
 
+// ClearAllStateMetrics clears all state metrics.
+func (m *Metrics) ClearAllStateMetrics() {
+	m.RecordChannelQueueLength(0)
+	atomic.StoreInt64(&m.pendingDABytes, 0)
+	m.pendingBlocksBytesCurrent.Set(0)
```

Comment:
This should cover any metric which is a `Gauge` and is incremented / decremented rather than "set". 

Counter Metrics only ever go up, so they can't be reset and shouldn't be. 

Gauge Metrics which are "set" will get the right value the next time they are updated and don't strictly need to be reset to zero -- although it might still be a good idea to to that.

Reply:
maybe worth adding this to the godoc of this function, so if any new gauges are added in the future, there's a reminder to add them here.

Reply:
added here [ddc5991](https://github.com/ethereum-optimism/optimism/pull/14780/commits/ddc5991fc6028c2d8f7421b758e74d12f249a139)